### PR TITLE
Feature/2562/embed bar charts election map homepage

### LINKED ISSUE #2562: Add election search and bar charts to homepage
**What we're after:** Following on the design work done [here](https://github.com/fecgov/fec-cms/issues/2674), we need to now implement the design on the front-end. Specifically, we need to add the election search to the FEC.gov homepage. As part of this work, we are also making tweaks to the language in the Getting started section of the page. 

**Completion criteria:**
- [x] Incorporate the election search map into the gray box on the homepage
- [x] Change header above election search that says "Compare candidates in elections"
- [x] Include copy in Compare candidate section: "Learn about candidates running in a particular state or district for past or upcoming elections."
- [x] Create new section between election search map and bar charts for these links (next item), use icon and text format, as seen in website menus. Separate with white lines. 
- [x] Include links to citizen guide sections
  - 1: Understand ways to support federal candidates https://www.fec.gov/introduction-campaign-finance/understanding-ways-support-federal-candidates/. Use direction sign icon.
  - 2: Learn more about using the FEC's campaign finance data https://www.fec.gov/introduction-campaign-finance/how-to-research-public-records/. Use financial document icon.
  - 3: Find election and voting information https://www.fec.gov/introduction-campaign-finance/election-and-voting-information/. Use election icon.
- [x] Change "Getting started" header to "Who's raising and spending the most"
- [x] Include copy: "Learn how much individual candidates have raised and spent for presidential, Senate and House elections."
- [x] Include modified version of the raising and spending bar charts in toggle format so that users can switch between the two. 
- [x] Add interactions to search the bar chart data including toggle, and two-single select dropdowns for the candidate selection and election year selection. 
- [x] Show top three raisers/spenders in bar chart

### Homepage design

screen shot 2019-02-22 at 4 28 41 pm

### PR DESCRIPTION
**Pull request by johnnyporkchops**
## Summary
Embed bar charts and election map in homepage

- Resolves #2562
- Replaces WIP PR: #2715

Please not that this iteration of the home `/templates/home/home_page.html` extends  a new template called `home_base.html` instead of the  `base.html` which is extended by most of the other pages in home. This is done in order to Set the `BASE_PATH` variables  to `/data` ( it is set to  `/ ` in `base.html` )

## Impacted areas of the application
List general components of the application that this PR will affect:

**Modifies files**:    
        modified:   /settings/base.py
	modified:   fec/static/js/modules/top-entities.js
	modified:   fec/static/scss/components/_breakdowns.scss
	modified:   fec/static/scss/components/_icon-headings.scss
	modified:   fec/static/scss/components/_search-controls.scss
	modified:   fec/static/scss/components/_table-styles.scss
	modified:   fec/static/scss/layout/_slabs.scss
	modified:   home/templates/home/home_page.html

**New Files:**
	fec/templates/home_base.html
	home/templates/partials/elections-lookup.html
	home/templates/partials/raising-spending.html
	home/templatetags/elections_lookup.py
	home/templatetags/top_entities.py

## Screenshots

<kbd>

embeds_scrnsht

</kbd>

## Related PRs
List related PRs against other branches:

## How to test
Include any information that may be helpful to the reviewer(s).
  - Run npm `run-build`
 - Set feature flags in your shell session or in your bash profile to map and barcharts:
 `FEC_FEATURE_HOME_BARCHARTS=1` , `FEC_FEATURE_HOME_MAP=1`
- Visit http://127.0.0.1:8000/ and test the embedded map and barchart apps
- This branch is based on the new develop branch with node upgrade, so if you have not pulled the new develop (and updated your environment), you may need to follow some of the how-to-test  steps here  to get this to work:  https://github.com/fecgov/fec-cms/pull/2733

**For future reference, should we need to make data apps available to home templates:**
Because the data app uses jinja ( .jinja) templates and the home app uses django (.html) templates, making the bar-charts and election-map available to the homepage involves recreating the logic in views/macros in data as templatetags/inclusion-tags in home. There are other ways of doing this, but I am fairly confident that this is the most efficient way. 

We have discussed making the templates uniform, using either Jinja or  Django templates sitewide. I cannot remember which was the recommendation by @xtine at  18F. Once this is done, sharing content across the apps would be simpler.

### CODE REVIEW

File: fec/home/templates/partials/raising-spending.html [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        {% for datum in data %}
+        <div role="row" class="js-top-row simple-table__row">
+
+        </div>
+        {% endfor %}
+        </div>
+
+      <div class="content__section row u-margin--top">
+        <ul class="list--buttons u-float-right">
+          <li><a class="button button--standard button--table js-browse"  href="/data/receipts">Explore top <span>raising</span> candidates</a></li>
```
Comment (patphongs) [inline fec/home/templates/partials/raising-spending.html]: This should be linking to `/data/raising-bythenumbers/` for top raising candidates

File: fec/fec/static/js/modules/top-entities.js [hunk 2]
```diff
@@ -245,4 +248,34 @@ TopEntities.prototype.pushStateToURL = function(keyValPairsObj) {
...
+  };
+
+  this.currentQuery = baseQuery;
+
+  this.loadData(this.currentQuery);
+  this.updateCoverageDateRange();
+
+  $('a.js-browse')
+    .attr({
+      href: '/data/' + this.type
```
Comment (patphongs) [inline fec/fec/static/js/modules/top-entities.js]: This should be linking to raising-bythenumbers or spending-bythenumbers pages

File: fec/home/templates/partials/raising-spending.html [hunk 3]
```diff
@@ -0,0 +1,91 @@
+{% if FEATURES.barcharts %}
+{% load wagtailcore_tags %}
+{% load filters %}
+
+<h2 class="heading--section u-margin--bottom">Who is raising and spending the most</h2>
+<section class="usa-grid-full">
+  <p class="t-bold">Learn how much individual candidates have raised and spent for presidential, Senate and House elections.</p>
```
Comment (patphongs) [inline fec/home/templates/partials/raising-spending.html]: `presidential, senate, and house` should be lowercased

File: fec/home/templates/partials/raising-spending.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        {% for datum in data %}
+        <div role="row" class="js-top-row simple-table__row">
+
+        </div>
+        {% endfor %}
+        </div>
+
+      <div class="content__section row u-margin--top">
+        <ul class="list--buttons u-float-right">
+          <li><a class="button button button--go js-browse"  href="/data/raising-bythenumbers/">Explore top <span>raising</span> candidates</a></li>
```
Comment (patphongs) [inline fec/home/templates/partials/raising-spending.html]: On hover, it's applying the blue underline under the button. That should be removed.

Screen Shot 2019-03-18 at 1 27 15 PM

File: fec/home/templates/home/home_page.html [hunk 5]
```diff
@@ -145,9 +151,26 @@ <h1>Commissioners</h1>
...
 {% block extra_js %}
-  <script type="text/javascript" src="{% asset_for_js 'home.js' %}"></script>
+<script src="{% asset_for_js 'bythenumbers.js' %}"></script>
+<script src="{% asset_for_js 'dataviz-common.js' %}"></script>
+<script src="{% asset_for_js 'election-lookup.js' %}"></script>
+<script>
+  //remove competing/cofusing querystrings on homepage
+  $(function(){
+    function cleanURI(){
+      let uri = window.location.toString();
```
Comment (rfultz) [inline fec/home/templates/home/home_page.html]: Could this be a const? Obviously not a big concern

File: fec/home/templates/home/home_page.html [hunk 6]
```diff
@@ -145,9 +151,26 @@ <h1>Commissioners</h1>
...
+      const clean_uri = uri.substring(0, uri.indexOf("?"));
+      if (uri.indexOf("?") > 0) {
```
Comment (rfultz) [inline fec/home/templates/home/home_page.html]: We should probably single-quote these two lines for consistency

Comment (johnnyporkchops) [reply fec/home/templates/home/home_page.html]: Good call, will do.

File: fec/home/templates/home/home_page.html (hunk 5 above)
Comment (johnnyporkchops) [reply fec/home/templates/home/home_page.html]: Yes, I think that makes sense after re-educating myself on let vs const.

File: fec/home/templates/partials/raising-spending.html (hunk 3 above)
Comment (johnnyporkchops) [reply fec/home/templates/partials/raising-spending.html]: per Jonella's comment, we decided to leave these as is.

**Comment by JonellaCulmer** [comment]
@johnnyporkchops I've annotated a screenshot with my feedback. Please see below:
Homepage_review

**Comment by johnnyporkchops** [comment]
@jonella, I have made the design edits changes you requested to the closes approximation I can within the current structure. 
In terms of stacking the bars and numbers--the best way to do this would probably be to create a new handlebars template for this but we are trying to use the same assets across data and home for this It might be possible with CSS, but I do not wan't to hold up this PR while I am digging into that.

**Pull request merged**